test/Mono.Linker.Tests.Cases/Reflection/ObjectGetType.cs

```diff
@@ -4,61 +4,792 @@
 
 namespace Mono.Linker.Tests.Cases.Reflection
 {
+	[ExpectedNoWarnings]
 	public class ObjectGetType
 	{
 		public static void Main ()
 		{
-			TestSealed ();
-			TestUnsealed ();
+			SealedType.Test ();
+			UnsealedType.Test ();
+
+			BasicAnnotationWithNoDerivedClasses.Test ();
+			MultipleValuesWithAnnotations.Test (0);
+			MultipleValuesWithAnnotations.Test (1);
+			MultipleValuesWithAndWithoutAnnotationsWarns.Test (0);
+			MultipleValuesWithAndWithoutAnnotationsWarns.Test (1);
+			MultipleValuesWithAndWithoutAnnotationsWarns.Test (2);
+
+			SingleDerivedWithAnnotatedParent.Test ();
+			DerivedWithBaseAndAnnotatedInterface.Test ();
+			DeepHierarchy.Test ();
+			DeepInterfaceHierarchy.Test ();
+
+			ConstructorAsSource.Test ();
+
+			InterfaceSeenFirst.Test ();
+			AnnotationsRequestedOnImplementation.Test ();
+			AnnotationsRequestedOnInterface.Test ();
+
+			AllAnnotationsAreApplied.Test ();
+			SealedWithAnnotation.Test ();
 		}
 
 		[Kept]
-		static void TestSealed ()
+		class SealedType
 		{
-			s_sealedClassField = new SealedClass ();
-			s_sealedClassField.GetType ().GetMethod ("Method");
+			[Kept]
+			static SealedClass s_sealedClassField;
+
+			[Kept]
+			sealed class SealedClass
+			{
+				[Kept]
+				public SealedClass () { }
+
+				[Kept]
+				public static void Method () { }
+
+				public static void UnusedMethod () { }
+			}
+
+			[Kept]
+			public static void Test ()
+			{
+				s_sealedClassField = new SealedClass ();
+				s_sealedClassField.GetType ().GetMethod ("Method");
+			}
 		}
 
 		[Kept]
-		[UnrecognizedReflectionAccessPattern (typeof (Type), nameof (Type.GetMethod), new Type[] { typeof (string) }, messageCode: "IL2075")]
-		static void TestUnsealed ()
+		class UnsealedType
 		{
-			s_unsealedClassField = new UnsealedClass ();
+			[Kept]
+			static UnsealedClass s_unsealedClassField;
 
-			// GetType call on an unsealed type is not recognized and produces a warning
-			s_unsealedClassField.GetType ().GetMethod ("Method");
+			[Kept]
+			class UnsealedClass
+			{
+				[Kept]
+				public UnsealedClass () { }
+
+				public static void Method () { }
+			}
+
+			[Kept]
+			[ExpectedWarning ("IL2075", "GetMethod")]
+			public static void Test ()
+			{
+				s_unsealedClassField = new UnsealedClass ();
+
+				// GetType call on an unsealed type is not recognized and produces a warning
+				s_unsealedClassField.GetType ().GetMethod ("Method");
+			}
 		}
 
 		[Kept]
-		static SealedClass s_sealedClassField;
+		class BasicAnnotationWithNoDerivedClasses
+		{
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
+			public interface IBasicAnnotatedInterface
+			{
+			}
+
+			[Kept]
+			[KeptMember (".ctor()")]
+			class ClassImplementingAnnotatedInterface : IBasicAnnotatedInterface
+			{
+				[Kept]
+				public void UsedMethod () { }
+				[Kept] // The type is not sealed, so trimmer will apply the annotation from the interface
+				public void UnsedMethod () { }
```

Comment:
Nit: Unsed instead of Unused in all the test inside BasicAnnotationWithNoDerivedClasses